docs/documentation/performance.md

```diff
@@ -1,74 +1,231 @@
+| Data Faker | openjdk-11.0.15.0.10-1.fc36.x86_64         | thrpt | 10  | 1057.132 ± 27.611 | ops/ms |
+| Data Faker | openjdk-18.0.1.0.10-1.rolling.fc36.x86_64  | thrpt | 10  | 1113.307 ± 21.496 | ops/ms |
+| Java Faker | openjdk-1.8.0.332.b09-1.fc36.x86_64        | thrpt | 10  | 145.144 ± 2.523   | ops/ms |
+| Java Faker | openjdk-11.0.15.0.10-1.fc36.x86_64         | thrpt | 10  | 147.812 ± 2.206   | ops/ms |
+| Java Faker | openjdk-18.0.1.0.10-1.rolling.fc36.x86_64  | thrpt | 10  | 207.275 ± 3.522   | ops/ms |
+
+5. Similar example as previous, however there are 3 methods. Besides, cache of parsed methods there was also added cache for parsed args.
+
+=== "Java"
+
+```java
+    @Benchmark
+    @BenchmarkMode(Mode.Throughput)
+    public void threeMethodsCallExpression(Blackhole blackhole) {
+        blackhole.consume(FAKER.expression("#{Name.first_name} #{Name.first_name} #{Name.last_name}"));
+    }
+```
+
+| Subject    | Java Version                               | Mode  | Cnt | Score            | Units  |
+|:-----------|:-------------------------------------------|:------|:----|:-----------------|:-------|
+| Data Faker | openjdk-1.8.0.332.b09-1.fc36.x86_64        | thrpt | 10  | 276.175 ± 6.503  | ops/ms |
+| Data Faker | openjdk-11.0.15.0.10-1.fc36.x86_64         | thrpt | 10  | 321.559 ± 12.098 | ops/ms |
+| Data Faker | openjdk-18.0.1.0.10-1.rolling.fc36.x86_64  | thrpt | 10  | 334.248 ± 12.733 | ops/ms |
+| Java Faker | openjdk-1.8.0.332.b09-1.fc36.x86_64        | thrpt | 10  | 33.567 ± 0.515   | ops/ms |
+| Java Faker | openjdk-11.0.15.0.10-1.fc36.x86_64         | thrpt | 10  | 36.055 ± 0.460   | ops/ms |
+| Java Faker | openjdk-18.0.1.0.10-1.rolling.fc36.x86_64  | thrpt | 10  | 49.014 ± 1.994   | ops/ms |
+
+In conclusion, we could say that Data Faker (1.4.0) is faster in every test made here than Java Faker. However, it makes sense to keep in mind that these tests do not cover all possible use cases and could be considered only as a starting point for analysis.
+
+## Data Faker (1.4.0) vs Kotlin-faker (1.11.0)
+There is also [Kotlin-faker](https://github.com/serpro69/kotlin-faker). And it has a [performance comparison table](https://github.com/serpro69/kotlin-faker) in README page.
+Unfortunately there is not so much information what kind of benchmarks have been executed. After some search across the project I was able to find only [this page](https://github.com/serpro69/kotlin-faker/blob/master/docs/src/orchid/resources/pages/faker-comparisons.md)
+So, it looks like the only test which were done is checking `Faker.name().name()` performance. Ok, let's start with the similar test here. Similar, because we are going to use JMH which was not used in their test.
+If applicable we try to execute same tests we did for previous section. So let's start with the original test from Kotlin Faker
+
+Original Test:
+=== "Java"
+
+```java
+    @Benchmark
+    @BenchmarkMode(Mode.Throughput)
```

Comment:
No results of this test?

Reply:
in the table below.
The only reason I put here 2 tests in one is that they have different clasnames and `getName`/`name` methods
like `io.github.serpro69.kfaker.Faker faker = new io.github.serpro69.kfaker.Faker();` vs `net.datafaker.Faker faker = new net.datafaker.Faker();` and `faker.getName().name();` vs `faker.name().name();`